packages/server/src/restaurants/dishes/domain/DeleteDishHandler.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+
+import { Handler } from '../../../shared';
+import { Dish, DishDocument } from '../database';
+
+interface DeleteDishRequest {
+  readonly id: string;
+  readonly restaurantId: string;
+}
+
+@Injectable()
+class DeleteDishHandler implements Handler<DeleteDishRequest, null | undefined> {
+  constructor(@InjectModel(Dish.name) private dishModel: Model<DishDocument>) {}
+
+  async exec(req: DeleteDishRequest): Promise<null | undefined> {
+    const result = await this.dishModel.deleteOne({ _id: req.id, restaurant: req.restaurantId });
+    if (result.deletedCount !== 1) return null;
+    return undefined;
```

Comment:
CRUDowo jest spoko, ale z tego co widzę brakuje dwóch rzeczy bardziej z logiki biznesowej:
1. Ten TODO z góry: `// TODO: Also remove dish from restaurant's dish array!!!`, obecnie (chyba bo nie testowałem) usunie się dish z kolekcji Dish, ale restauracja w której było to danie dalej będzie miało jego id na liście. (myślę, że [to](https://docs.mongodb.com/manual/reference/operator/update/pull/) może Ci się przydać)
2. Nie ma sprawdzenia właściciela dania (sprawdzamy w kontrolerze, czy user jest partnerem, ale nigdy czy to danie należy do niego), więc "Restauracja 1" może bezkarnie usuwać dania "Resturacji 2". (pobrać dish i porównać id `restaurant` z id przekazanym do kontrolera przez `@PartnerId()`)

Reply:
2. Tutaj nie jestem pewna czy rozumiem. Jeżeli nie wyszuka mi takiego `deleteOne` jak chciałam czyli o `restaurantId` zalogowanego z autoryzaji oraz `id` (dania) to wyrzuca mi 404 bo np. nie zgadza się  `restaurantId`  . To wydaje mi się, że załatwia ten case właśnie o którym wpsomniałeś.

Reply:
Prawda, przeoczyłem to.